core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/OnDemandRepairJob.java

```diff
@@ -115,7 +118,12 @@ public RepairConfiguration getRepairConfiguration()
 
     public RepairJobView getView()
     {
-        return new RepairJobView(getId(), myTableReference, myRepairConfiguration, null);
+        return new RepairJobView(getId(), myTableReference, myRepairConfiguration, null, getStatus(), getProgress());
+    }
+
+    private RepairJobView.Status getStatus()
+    {
+        return failed ? RepairJobView.Status.ERROR : RepairJobView.Status.IN_QUEUE;
```

Comment:
If the token ranges changes before we have had a change to run this will cause the job to be removed from the queue with an appropriate error message but the status of the job (according to RepairJobView) will still be IN_QUEUE.